src/ServiceMonitor/IISConfigUtil.cpp

```diff
@@ -257,37 +260,44 @@ HRESULT IISConfigUtil::UpdateEnvironmentVarsToConfig(WCHAR* pstrAppPoolName)
                 
                 continue;
             }
-            hr = BuildAppCmdCommand(pstrName, pstrValue, pstrAppPoolName, &pstrAddCmd, TRUE);
-            if (FAILED(hr))
-            {
-                goto Finished;
-            }
+			wstring * pStrTempName = new wstring();
+			pStrTempName->append(pstrName);
+			wstring * pStrTempValue = new wstring();
+			pStrTempValue->append(pstrValue);
 
-            hr = BuildAppCmdCommand(pstrName, pstrValue, pstrAppPoolName, &pstrRmCmd, FALSE);
-            if (FAILED(hr))
-            {
-                goto Finished;
-            }
+			envSet.insert(KV_WSTR(*pStrTempName, *pStrTempValue));
+            
             pEqualChar[0] = L'=';
 
-
-            //allow appcmd to fail if it is trying to remove environment variable
-            RunCommand(pstrRmCmd);
-
-            //appcmd must success when add new environment variable
-            hr = RunCommand(pstrAddCmd);
-
-            if (FAILED(hr))
-            {
-                goto Finished;
-            }
-        }
+		}
         //
         // move to next environment variable
         //
         lpszVariable += lstrlen(lpszVariable) + 1;
     }
 
+	hr = BuildAppCmdCommand(envSet, pstrAppPoolName, &pstrAddCmd, TRUE);
+	if (FAILED(hr))
+	{
+		goto Finished;
+	}
+
+	hr = BuildAppCmdCommand(envSet, pstrAppPoolName, &pstrRmCmd, FALSE);
+	if (FAILED(hr))
+	{
+		goto Finished;
+	}
+
+	//allow appcmd to fail if it is trying to remove environment variable
+	RunCommand(pstrRmCmd, TRUE);

```

Comment:
Was this behavior already discussed?

Reply:
When do docker stop and docker start. ServiceMonitor.exe will run again and try to add duplicated Env Var to AppHost.config. So every time we run ServiceMonitor.exe, we will try to remove the Env Var first. This command will fail if AppHost.config does not contain Env Var. But we will ignore the failure.